chain/chain/src/types.rs

```diff
@@ -699,6 +699,7 @@ pub trait RuntimeAdapter: Send + Sync {
         request: &QueryRequest,
     ) -> Result<QueryResponse, near_chain_primitives::error::QueryError>;
 
+    /// WARNING: this call may be expensive.
```

Comment:
(only if you're asking about the block and not the epoch)

Reply:
This is just a trait. It is up to implementation what is expensive and what not. This warning is just to make the reader careful.